### PR TITLE
Feature/transitional value

### PR DESCRIPTION
**Pull request by DovydasNavickas**
Transitional value for modifiers.

### CODE REVIEW

File: packages/simplr-forms/src/modifiers/string-to-decimal.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+export class StringToDecimalModifier extends BaseModifier<StringToDecimalProps, {}> {
+    static defaultProps: StringToDecimalProps = {
+        delimiter: "."
+    };
+
+    private static emptyFormattedValue: string;
+
+    Format(value: FieldValue): FieldValue {
+        console.log(value);
```
Comment (MartynasZilinskas) [inline packages/simplr-forms/src/modifiers/string-to-decimal.ts]: Remove this console.log

File: packages/simplr-forms/src/modifiers/string-to-decimal.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        if (value !== EMPTY_VALUE) {
+            return value.toString();
+        }
+        return "";
```
Comment (MartynasZilinskas) [inline packages/simplr-forms/src/modifiers/string-to-decimal.ts]: Maybe return EMPTY_VALUE

File: packages/simplr-forms/src/modifiers/string-to-decimal.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                    return {
+                        Value: 0,
+                        TransitionalValue: "-"
+                    };
+                }
+            }
+
+            const leadingMinus = negative ? "-" : "";
+            const regex = new RegExp(`[^0-9\\${this.props.delimiter}]+`, "g");
+            console.log(`Delimiter: ${this.props.delimiter}`);
```
Comment (MartynasZilinskas) [inline packages/simplr-forms/src/modifiers/string-to-decimal.ts]: Remove this console.log

File: packages/simplr-forms/src/modifiers/string-to-decimal.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            let transitionalValue = this.TrimLeft(extractedValue, "0");
+
+            // Add leading zero, if fraction is being entered
+            if (transitionalValue.substr(0, 1) === this.props.delimiter) {
+                transitionalValue = "0" + transitionalValue;
+            }
+
+            if (this.props.precision != null) {
+                const delimiterIndex = transitionalValue.indexOf(this.props.delimiter!);
+                console.log(`Delimiter index: ${delimiterIndex}`);
```
Comment (MartynasZilinskas) [inline packages/simplr-forms/src/modifiers/string-to-decimal.ts]: Remove this console.log

File: packages/simplr-forms/src/utils/value-helpers.ts [hunk 5]
```diff
@@ -67,10 +75,14 @@ export function ProcessValue<TProcessor, TProcessedResult>(
 
     const renderedProcessors = RenderComponents<TProcessor>(processors);
 
+    let processedValue: TProcessedValue | undefined = undefined;
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms/src/utils/value-helpers.ts]: Default is `undefined`

File: packages/simplr-forms/src/modifiers/string-to-decimal.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+            let transitionalValue = this.TrimLeft(extractedValue, "0");
+
+            // Add leading zero, if fraction is being entered
+            if (transitionalValue.substr(0, 1) === this.props.delimiter) {
+                transitionalValue = "0" + transitionalValue;
+            }
+
+            if (this.props.precision != null) {
+                const delimiterIndex = transitionalValue.indexOf(this.props.delimiter!);
```
Comment (GiedriusGrabauskas) [inline packages/simplr-forms/src/modifiers/string-to-decimal.ts]: `delimiter` move to variable

**Pull request merged**